src/percy-agent-client/serialize-cssom.ts

```diff
@@ -0,0 +1,30 @@
+// Take all the CSS created in the CSS Object Model (CSSOM), and inject it
+// into the DOM so Percy can render it safely in our browsers.
+// Design doc:
+// https://docs.google.com/document/d/1Rmm8osD-HwSHRpb8pQ_1wLU09XeaCV5AqMtQihk_BmM/edit
+export function serializeCssOm(document: HTMLDocument) {
+  [].slice.call(document.styleSheets).forEach((styleSheet: CSSStyleSheet) => {
+    // Make sure it has a rulesheet, does NOT have a href (no external stylesheets),
+    // and isn't already in the DOM.
+    const hasHref = styleSheet.href
+    const ownerNode = styleSheet.ownerNode as HTMLElement
+    const hasStyleInDom = ownerNode.innerText && ownerNode.innerText.length > 0
+
+    if (!hasHref && !hasStyleInDom && styleSheet.cssRules) {
+      const serializedStyles = [].slice
+        .call(styleSheet.cssRules)
+        .reduce((prev: string, cssRule: CSSRule) => {
+          return prev + cssRule.cssText
+        }, '')
+
+      // Append the serialized styles to the styleSheet's ownerNode to minimize
+      // the chances of messing up the cascade order.
+      const serializedSheet = document.createElement('style')
+      serializedSheet.type = 'text/css'
```

Comment:
it would be good to be able to identify this CSS as something _we_ injected in. Initially we'd talked about a data attribute or some type but I think [Percy specific CSS](https://docs.percy.io/docs/percy-specific-css) would be a particularly nice solution.

I guess it depends on what `serializedStyles` looks like. Could the serialized styles be simply wrapped in a `@media` block like this?

```css
@media only percy {
  /** CSSOM styles here **/
}
```

Reply:
If we can't get that `@media` thing to work, since I'm wrapping this style sheet in an element I'm creating myself, I could add a custom attribute to it, e.g. `data-percy-serialized` or something like that.

Reply:
Ok, added `data-percy-cssom-serialized` to the serialized styles!